scripts/Muon/GUI/FrequencyDomainAnalysis/MaxEnt/maxent_presenter_new.py

```diff
@@ -0,0 +1,169 @@
+
+    # constructs the inputs for the MaxEnt algorithms
+    # then executes them (see maxent_model to see the order
+    # of execution
+    def handleMaxEntButton(self):
+        # put this on its own thread so not to freeze Mantid
+        self.thread = self.createThread()
+        self.thread.threadWrapperSetUp(self.deactivate, self.handleFinished)
+
+        # make some inputs
+        inputs = {}
+
+        if self.view.usePhases() and self.view.calcPhases():
+            inputs["phaseTable"] = self.get_phase_table_inputs()
+
+        inputs["maxent"] = self.getMaxEntInput()
+
+        inputs["Run"] = self.get_input_run()
+
+        self.thread.loadData(inputs)
+        self.thread.start()
+
+    def get_phase_table_inputs(self):
+        phaseTable = {}
+        phaseTable["FirstGoodData"] = self.view.getFirstGoodData()
+        phaseTable["LastGoodData"] = self.view.getLastGoodData()
+        phaseTable["InputWorkspace"] = self.view.getInputWS()
+        phaseTable["DetectorTable"] = "PhaseTable"
+        phaseTable["DataFitted"] = "fits"
+
+        return phaseTable
+
+    def get_input_run(self):
+        return self.view.getInputWS().split("_", 1)[0]
+
+    # kills the thread at end of execution
+    def handleFinished(self):
+        self.activate()
+        self.thread.deleteLater()
+        self.thread = None
+        self.updatePhaseOptions()
+
+    def updatePhaseOptions(self):
+        inputs = {}
+        self.view.addOutputPhases(inputs)
+        name = inputs['OutputPhaseTable']
+        name = name[1:]
+
+        current_list = self.view.getPhaseTableOptions()
+        if self.phaseTableAdded(name) and name not in current_list:
+            index = self.view.getPhaseTableIndex()
+            self.view.addPhaseTableToGUI(name)
+            self.view.setPhaseTableIndex(index)
+
+    def phaseTableAdded(self, name):
+        return mantid.AnalysisDataService.doesExist(name)
+
+    def getMaxEntInput(self):
+        inputs = self.view.initMaxEntInput()
+
+        if self.view.outputPhases():
+            self.view.addOutputPhases(inputs)
+
+        if self.view.outputDeadTime():
+            self.view.addOutputDeadTime(inputs)
+
+        if self.view.outputPhaseEvo():
+            self.view.addOutputPhaseEvo(inputs)
+
+        if self.view.outputTime():
+            self.view.addOutputTime(inputs)
+
+        # for new version 2 of FDA
+        self.cleanOutputsForVersion2(inputs)
+
+        if self.view.usePhases():
+            self.view.addPhaseTable(inputs)
+
+        return inputs
+
+    def cleanOutputsForVersion2(self, inputs):
+        inputs["InputWorkspace"] = self.view.getInputWS()
+        keys = [
+            "OutputWorkspace",
+            "OutputPhaseTable",
+            "OutputDeadTimeTable",
+            "PhaseConvergenceTable",
+            "ReconstructedSpectra"]
+        for output in keys:
+            if output in inputs:
+                inputs[output] = inputs[output][1:]
```

Comment:
sorry, I dont understand what/why we are doing this

Reply:
I'm not quite sure either. This is some legacy code which is removed in #25573 however so can we leave it as is for now.